gradle/dependencies.gradle

```diff
@@ -1,5 +1,5 @@
 ext {
-  def kotlinVersion = "1.5.31"
+  def kotlinVersion = "1.7.10"
```

Comment:
How do you decide when to bump? It affects the apps, right? It may force them to bump too. Do you have some statistics on who uses which version?

Reply:
In theory, unless it's a major version this shouldn't impact downstream clients / apps, as won't bring breaking changes. We're not exposing Kotlin constructs either.

Reply:
we're never investigate what version of koltin uses upstream/downstream sdks/apps, worked with 1.x as it doesn't brings breaking changes.  

> It affects the apps, right? It may force them to bump too.

I don't think so 🤔 kotlin compiler compiles kt files to java binaries. What means new features just doesn't realised by newest version of kotlin, but it shouldn't break anything

Reply:
Kotlin comes in with breaking changes even for _minor_ releases (if we were looking at SEMVER), which they call a _feature_ release - https://kotlinlang.org/docs/kotlin-evolution.html#feature-releases-and-incremental-releases:
> Feature releases often add new features and may remove or change previously deprecated ones. Feature graduation from pre-stable to stable also happens in feature releases.

Only SEMVER-like patches are targeting binary compatibility.

Reply:
interesting...

from https://kotlinlang.org/docs/kotlin-evolution.html#compatibility-flags
>To give you more time for migration, we [support](https://kotlinlang.org/docs/compatibility-modes.html) the development for three previous language and API versions in addition to the latest stable one.\

so we're good to use 1.7.x which support +1.4.0 features

anyway, lat's discuss it cross-team as we already plan

Reply:
But we're not directly exposing any of these features directly through the SDK. In any case, you never know what Kotlin features downstream clients / apps are relying on so there's no way (at least I can't think of any) to prevent this kind of issues from our side other than documenting pretty well publicly and in advance our release train with their respective dependencies - including Kotlin versions. This is actually an on-going discussion cc @zugaldia

Reply:
> publicly and in advance our release train with their respective dependencies - including Kotlin versions.

agree 👍